### ISSUE TITLE
Reimport detection

### ISSUE THREAD

**Issue by spaceby** [astral-sh/ruff#20131]
### Summary

I'm interested in the situation where a method is imported from a file that imports it from another file

For example

file_with_function.py
```python
def foo():
    pass
```
---
file_with_import.py
```python
from file_with_function import foo
```
---
main.py
```python
from file_with_import import foo
```
---

Reimports are hard to spot with the naked eye. They can't be found by searching. And they can cause the hard-to-detect circular import problem.

A rule to correct this would be useful.

**Comment by ntBre** [comment]
I think this is the same as #19559, so I'll close this as a duplicate. Thanks for the suggestion!

https://github.com/astral-sh/ruff/issues/20092 is another related issue
